lib/Core/ServerConfig.d.ts

```diff
@@ -1,6 +1,16 @@
+export declare interface ServerConfigOptions {
+  version: string;
+  proxyAllDomains: boolean;
+  allowProxyFor: string[];
+  maxConversionSize: number;
+  newShareUrlPrefix?: string;
+  shareUrlPrefixes: object;
+  additionalFeedbackParameters: object[];
+}
+
 declare class ServerConfig {
-  config: unknown;
-  init(serverConfigUrl: string): Promise<unknown>;
+  config: ServerConfigOptions;
```

Comment:
It's not for this PR, but I find the ServerConfig design a bit puzzling, the object caches a configuration, but `ServerConfig.init()` that uses the cached configuration is only called from `Terria.start()`, and that call is always done directly after `new ServerConfig()`. To me, it seems it would have been easier to have `ServerConfig.init()` be the equivalent of a static function in other languages so ServerConfig object would never have to be allocated at all.

Reply:
This way server endpoint will be called only once, or do you mean about the server config stored in the terria class?

Reply:
My point is that terria.start() looks like this:
```typescript
this.serverConfig = new ServerConfig();
const serverConfig = this.serverConfig.init(...);
```
so init is only called immediately after new, and the cache will never be used. The ServerConfig object could just as well have been omitted and the code looked like this instead:
```typescript
this.serverConfigOptions = mkServerConfigOptions(...);
```
which would omit one level of indirection (and save a few bytes of object allocation), and otherwise have provided the same performance characteristics.